src/index/updater.rs

```diff
@@ -98,7 +98,7 @@ impl<'index> Updater<'index> {
 
       uncommitted += 1;
 
-      if uncommitted == 5000 {
+      if uncommitted >= self.index.settings.commit_interval {
```

Comment:
Since `uncommitted` increases by 1 each time, I think `==` is clearer:

```suggestion
      if uncommitted == self.index.settings.commit_interval {
```